nautobot/core/api/utils.py

```diff
@@ -322,11 +322,49 @@ class Meta:
     return field_class, field_kwargs
 
 
+BRIEF_NESTED_SERIALIZER_CACHE = {}
+
+
+def brief_nested_serializer_factory(relation_info):
+    """
+    Return a BriefSerializer representation of a serializer field.
+    This method should only be called in build_nested_field()
+    in which relation_info is are already given.
+    """
+    brief_serializer_name = f"BriefNested{relation_info.related_model.__name__}"
+    # If we already have built a suitable NestedSerializer we return the cached serializer.
+    # else we build a new one and store it in the cache for future use.
+    if brief_serializer_name in BRIEF_NESTED_SERIALIZER_CACHE:
+        field_class = BRIEF_NESTED_SERIALIZER_CACHE[brief_serializer_name]
+        field_kwargs = get_nested_relation_kwargs(relation_info)
+    else:
+        base_serializer_class = get_serializer_for_model(relation_info.related_model)
+
+        class NautobotBriefNestedSerializer(base_serializer_class):
+            def get_field_names(self, declared_fields, info):
+                """Ensure id, object_type, and url are only in the brief serializer."""
+                base_fields = list(super().get_field_names(declared_fields, info))
+                desired_fields = set(["id", "object_type", "url"])
+                return [field for field in base_fields if field in desired_fields]
```

Comment:
Because we use this method in our mixins to bolt on fields not defined in `Meta.fields` and we should inherit from the related models base serializer, we need to prevent them from carrying through.